owasp-top10-2017-apps/a5/ecommerce-api/app/handlers/handlers.go

```diff
@@ -13,14 +14,40 @@ func HealthCheck(c echo.Context) error {
 	return c.String(http.StatusOK, "WORKING\n")
 }
 
+func ErrorMessage(c echo.Context) error {
+	return c.JSON(http.StatusBadRequest, map[string]string{"result": "error", "details": "Error finding this UserID."})
+}
+
 // GetTicket returns the userID ticket.
 func GetTicket(c echo.Context) error {
+
+	cookie, err := ReadCookie(c)
```

Comment:
It looks like `cookie` is coming from the user input but no sanitisation is being made. 🧐 

An attacker could send anything into his/her cookie and It will be called in `jwt.ParseWithClaims()`, right?

```sh
curl --cookie "sessionIDa5=AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA" http://localhost:8888/ticket/4b2d515b-7541-4365-84ca-0d593f0d199a
```

Reply:
Yes, the parser will verify if the cookie is valid and return error if the input is invalid, like: `AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA`.